### PR TITLE
Add sync firestore to schedule

### PR DESCRIPTION
**Pull request by ChaAbby**
This PR syncs data from firestore in `scheduled_streams` collection to scheduled stream cards in schedule screen.

### CODE REVIEW

File: lib/home/schedule/schedule.dart [hunk 1]
```diff
@@ -1,4 +1,7 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
```
Comment (nate-thegrate) [inline lib/home/schedule/schedule.dart:3]:
```suggestion
```

For some reason, I've noticed that the "widgets" library gets auto-imported a lot, even though everything in there is covered by "material".

**Review by nate-thegrate — COMMENTED** [review]
Holy cow, that was fast!

I'm not making a second PR this time, since this already looks really good (and also since I'm currently working on some stuff and would like to avoid conflicts down the line).

I just have a few bits of feedback, and then we should be good to go!

File: lib/home/schedule/schedule.dart [hunk 2]
```diff
@@ -22,140 +125,40 @@ class Schedule extends StatelessWidget {
...
-                        style: StyleText(color: Colors.black),
-                      ),
-                    ),
-                  ),
-                ],
-              ),
+          ),
+        ),
+        if (scheduledStreams['active']!.isEmpty) const Center(child: CircularProgressIndicator()),
+          for (final stream in scheduledStreams['active']!) stream,
```
[the remark this answers is not in the document]
Comment (ChaAbby) [reply lib/home/schedule/schedule.dart:141]: oo thanks for this!

File: lib/firebase/src/firestore.dart [hunk 3]
```diff
@@ -4,6 +4,7 @@ import 'package:thc/utils/app_config.dart';
 enum Firestore {
   streams,
   surveys,
+  scheduled_streams,
```
[the remark this answers is not in the document]
Comment (ChaAbby) [reply lib/firebase/src/firestore.dart]: Thanks for catching this.

File: lib/home/schedule/schedule.dart [hunk 4]
```diff
@@ -7,9 +10,109 @@ import 'package:thc/utils/navigator.dart';
...
+
+class Schedule extends StatefulWidget {
   const Schedule({super.key});
 
+  @override
+  State<Schedule> createState() => _ScheduleState();
+}
+
+class _ScheduleState extends State<Schedule> {
+  Map<String, List<Widget>> scheduledStreams = {
```
[the remark this answers is not in the document]
Comment (ChaAbby) [reply lib/home/schedule/schedule.dart]: Good point! I changed it to two lists in my last commit.

**Review by nate-thegrate — APPROVED** [review]
Fantastic work here!

**Pull request merged**